components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/handler/OrgClaimMgtHandler.java

```diff
@@ -282,8 +285,19 @@ private void handleUpdateLocalClaim(Event event) throws IdentityEventException {
                 String sharedOrganizationTenantDomain = getOrganizationManager().
                         resolveTenantDomain(organization.getId());
                 if (isExistingLocalClaimURI(localClaimURI, sharedOrganizationTenantDomain)) {
-                    getClaimMetadataManagementService().updateLocalClaim(new LocalClaim(localClaimURI,
-                            mappedAttributes, localClaimProperties), sharedOrganizationTenantDomain);
+                    String primaryUserStoreDomain = getPrimaryUserStoreDomain(tenantDomain);
+                    Iterator<AttributeMapping> iterator = mappedAttributes.iterator();
+                    while (iterator.hasNext()) {
+                        AttributeMapping attributeMapping = iterator.next();
+                        if (!attributeMapping.getUserStoreDomain().equals(primaryUserStoreDomain)) {
```

Comment:
```suggestion
                        if (!primaryUserStoreDomain.equals(attributeMapping.getUserStoreDomain())) {
```
since primaryUserStoreDomain has the value for sure. Otherwise this will cause NPE

Reply:
Fixed